### ISSUE TITLE
Provide a DB version change API

### ISSUE THREAD

**Issue by tofumatt** [localForage/localForage#117]
Re: #116 -- we don't currently handle upgrade events, even in drivers that have very good support for upgrades (i.e. IndexedDB). We don't really employ a schema, but maybe it would make sense to provide an API to allow people to do data migrations across versions of their DB?

Or is this out-of-scope for localForage, and something an app/framework should handle?

**Comment by augbog** [comment]
Stumbling on this as I was wondering if localForage provided a way to help handle upgrades -- I personally would be very interested in this especially if we were able to provide a way for localStorage and WebSQL a way to migrate. I see packages like [angular-localforage-migrations](https://github.com/psalaets/angular-localforage-migrations) that wouldn't be as necessary if this were the case.

My original instinct that localForage versioning would somehow hook into IndexedDB but #658 showed that wasn't the case (thankfully there has been healthy discussion about this :D). Would like to see more thoughts around this though as I feel it would be a very useful feature.

**Comment by tofumatt** [comment]
Unfortunately right now upgrades aren’t really handled well by localForage :-(

It’d be an awesome feature to add but we haven’t been able to get to it.

**Comment by thgreasi** [comment]
It should be easy to make an plain js migration plugin based on what the angular one already does. I imagine having a main LF core (this repo) and several satellite plugins under the other. It would be easy to implement independently, evaluate & reiterate. It would also ease evaluating the implementation & tracking user acceptance (eg npm DLs).

On the other hand I'm thinking about a `__localforage__system__` instance/db/store for some time now, holding other things like the used instances.

**Comment by augbog** [comment]
@thgreasi yeah I actually didn't mean to suggest that the migration plugins aren't good -- if that's the preferred route for people to incorporate migrations in their apps, I just feel that should be specified somewhere (essentially #116) 😛
